src/java_bytecode/java_object_factory.cpp

```diff
@@ -863,6 +853,17 @@ void java_object_factoryt::gen_nondet_struct_init(
         true,    // allow_null always true for sub-objects
         depth,
         substruct_in_place);
+
+      if(name=="length" &&
+          (class_identifier=="java.lang.String" ||
+           class_identifier=="java.lang.StringBuffer" ||
+           class_identifier=="java.lang.StringBuilder"))
+      {
+        exprt max_length=from_integer(
```

Comment:
Add assumption where string max length cannot be greater than max value of `me.type()`.